### PR TITLE
Workerslaunchingspeed -   Several patches to fix bin/tachyon-workers.sh to speed up launching of workers on a cluster.

### PR DESCRIPTION
**Pull request by chu11**
Several patches to fix bin/tachyon-workers.sh to speed up launching of workers on a cluster.

### CODE REVIEW

File: bin/tachyon-workers.sh [hunk 1]
```diff
@@ -22,9 +22,15 @@ TACHYON_LIBEXEC_DIR=${TACHYON_LIBEXEC_DIR:-$DEFAULT_LIBEXEC_DIR}
...
-  echo -n "Connection to $worker as $USER... "
-  ssh -o ConnectTimeout=5 -o StrictHostKeyChecking=no -t $worker $LAUNCHER $"${@// /\\ }" 2>&1
-  sleep 0.02
+  if [ -n "${TACHYON_SSH_FOREGROUND}" ]; then
+    echo -n "Connection to $worker as $USER... "
+    ssh -o ConnectTimeout=5 -o StrictHostKeyChecking=no -t $worker $LAUNCHER $"${@// /\\ }" 2>&1
+  else
+    ssh -o ConnectTimeout=5 -o StrictHostKeyChecking=no -t $worker $LAUNCHER $"${@// /\\ }" 2>&1 &
+  fi
+  if [ "$TACHYON_SLAVE_SLEEP" != "" ]; then
```
Comment (calvinjia) [inline bin/tachyon-workers.sh]: Should this be set in the env-template to the original 0.02 value?

Comment (chu11) [inline bin/tachyon-workers.sh]: You mean to preserve original behavior?  Or just to list it there for documentation?

Comment (calvinjia) [inline bin/tachyon-workers.sh]: Preserve the original behavior. Having the sleep will probably not affect most users?

Comment (chu11) [inline bin/tachyon-workers.sh]: No, it shouldn't affect most users.

Comment (calvinjia) [inline bin/tachyon-workers.sh]: Okay, in that case, do you mind having the new options defaulted to the original settings in the env-template? (TACHYON_SSH_FOREGROUND and TACHYON_SLAVE_SLEEP)

Comment (chu11) [inline bin/tachyon-workers.sh]: Was already working on it, it's in the now updated branch.

**Comment by chu11** [comment]
Updated branch to include environment variables in conf/tachyon-env.sh.template to lead to original behavior.

**Comment by calvinjia** [comment]
+1 LGTM

**Pull request merged**